File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.cpp
class Solution {
public:
    bool is_square(int n){
        double sqrt_n=sqrt(n);
        return sqrt_n==int(sqrt_n);
    }
    bool judgeSquareSum(int n) {
        if (n==0) return 1;
        while (n%4==0) n/=4;
        if (n%8==7) return 0;
        int sqrt_n=int(sqrt(n));
        if (is_square(n)) return 1;
        for(int i=1; i<=sqrt_n; i++)
        {
            int y=n-i*i;
            if (is_square(y)) return 1;
        }
        return 0;
    }
};

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.cs
public class Solution {
        public bool IsSquare(int n) {
        double sqrtN = Math.Sqrt(n);
        return sqrtN == (int)sqrtN;
    }

    public bool JudgeSquareSum(int n) {
        if (n == 0) return true;
        while (n % 4 == 0) n /= 4;
        if (n % 8 == 7) return false;
        int sqrtN = (int)Math.Sqrt(n);
        if (IsSquare(n)) return true;
        for (int i = 1; i <= sqrtN; i++) {
            int y = n - i * i;
            if (IsSquare(y)) return true;
        }
        return false;
    }
}